File: programmers/level2/전화번호 목록/solution.js
function solution(clothes) {
  var answer = true;
  
  const hash = {};

  for (let i = 0; i < clothes.length; i++) {
    hash[clothes[i]] = clothes[i];
  }
  for (let i = 0; i < clothes.length; i++) {
    let temp = '';
    const cloth = [...clothes[i]];
    for (let j = 0; j < cloth.length; j++) {
      temp += cloth[j];
      if (hash[temp] && temp != clothes[i]) {
        answer = false;
        return answer;
      }
    }
  } 
  return answer;
}

File: programmers/level2/전화번호 목록/solution.py
def solution(phone_book):
    hash = {}
    
    for phone_number in phone_book:
        hash[phone_number] = phone_number
    
    for phone_number in phone_book:
        temp = ''
        for number in phone_number:
            temp += number
            if temp in hash and temp != phone_number:
                return False
        
    return True